### PR TITLE
Updating AccountPicker component to allow for customizations

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**
This PR refines the architecture of the `AccountPicker` component to reduce JSX and HTML bloat and leverages the `ButtonBase` props more effectively. It also introduces a new prop for custom className, enhancing the component's flexibility. Additionally, a new story has been added to demonstrate potential customizations.

## **Related Issues**
None

## **Manual Testing Steps**

1. Navigate to the Storybook build linked in this PR.
2. Use the search bar to find the `AccountPicker` component.
3. Verify that the existing AccountPicker stories remain unchanged. 
4. Check out the new story showcasing the customization possibilities with the updated component.

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/58b73652-04e2-47b1-86a3-d870981fbae8

Amount of HTML that was rendered before has some unnecessary wrapping divs `8` elements

Screenshot 2023-10-25 at 9 42 27 AM 

In app

Screenshot 2023-10-25 at 12 10 03 PM

### **After**
Added an example of how you can customize the AccountPicker

Screenshot 2023-10-25 at 8 53 42 AM

https://github.com/MetaMask/metamask-extension/assets/8112138/df18283c-5fdb-4f55-887a-cf538e515e27

Amount of HTML has been reduced to a total of `6` elements

Screenshot 2023-10-25 at 9 42 03 AM

In app

Screenshot 2023-10-25 at 12 07 51 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/account-picker/account-picker.js [hunk 1]
```diff
@@ -37,74 +37,68 @@ export const AccountPicker = ({
   addressProps = {},
   labelProps = {},
   textProps = {},
+  className = '',
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.js:40]: TODO: Add unit test to cover this prop

File: ui/components/multichain/account-picker/account-picker.js [hunk 2]
```diff
@@ -37,74 +37,67 @@ export const AccountPicker = ({
...
   ...props
 }) => {
   const useBlockie = useSelector(getUseBlockie);
   const shortenedAddress = shortenAddress(toChecksumHexAddress(address));
 
   return (
     <ButtonBase
-      className="multichain-account-picker"
+      className={classnames('multichain-account-picker', className)}
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.js:48]: Allows for additional className to be added

File: ui/components/multichain/account-picker/account-picker.js [hunk 3]
```diff
@@ -37,74 +37,67 @@ export const AccountPicker = ({
...
       textProps={{
         display: Display.Flex,
-        gap: 2,
         alignItems: AlignItems.center,
+        gap: 1,
         ...textProps,
       }}
       size={showAddress ? ButtonBaseSize.Lg : ButtonBaseSize.Md}
       disabled={disabled}
+      display={Display.Flex}
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.js:62]: May be able to remove this prop as it's already `Display.InlineFlex`  on `ButtonBase`
```suggestion
```

File: ui/components/multichain/account-picker/account-picker.stories.js [hunk 4]
```diff
@@ -55,3 +61,23 @@ export const ChaosWithAddressStory = (args) => (
...
+    labelProps={{ textAlign: TextAlign.Left }}
+    block
+    endIconProps={{
+      marginLeft: 'auto',
+    }}
+    textProps={{
+      gap: 2,
+    }}
+  />
+);
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.stories.js:83]: @NidhiKJha this is just an example you could possibly create a new component based on this

File: ui/components/multichain/account-picker/account-picker.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/multichain/account-picker/account-picker.js:36]: Done here https://github.com/MetaMask/metamask-extension/pull/21544/commits/95a974ce82afce3465209fd4a8eafae2b4669fc5

File: ui/components/multichain/account-picker/account-picker.stories.js [hunk 5]
```diff
@@ -11,19 +16,39 @@ export default {
...
+      control: 'object',
+    },
+    classNam: {
+      control: 'text',
+    },
   },
   args: {
     address: '0xb19ac54efa18cc3a14a5b821bfec73d284bf0c5e',
     name: 'Account 1',
-    onClick: () => undefined,
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.stories.js:26]: removing this makes actions work

https://github.com/MetaMask/metamask-extension/assets/8112138/9c3c1ff7-aa05-4b15-ae13-f4cbc944842e

File: ui/components/multichain/account-picker/account-picker.stories.js [hunk 6]
```diff
@@ -11,19 +16,39 @@ export default {
...
+    },
+    labelPros: {
+      control: 'object',
+    },
+    textProp: {
+      control: 'object',
+    },
+    classNam: {
+      control: 'text',
+    },
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.stories.js:47]: Adding missing argTypes

File: ui/components/multichain/account-picker/account-picker.test.js [hunk 7]
```diff
@@ -50,4 +50,9 @@ describe('AccountPicker', () => {
     const { getByText } = render({ showAddress: true });
     expect(getByText('0x0DCD5...3E7bc')).toBeInTheDocument();
   });
+
+  it('should allow for an additional class name via className prop', () => {
+    const { container } = render({ className: 'test-class' });
+    expect(container.querySelector('.test-class')).toBeInTheDocument();
+  });
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-picker/account-picker.test.js:57]: Adding test coverage for new className prop

**Review by NidhiKJha — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**